plugins/source/mysql/client/transformer.go

```diff
@@ -135,3 +140,41 @@ func (*Transformer) TransformTextArray(v *schema.TextArray) any {
 
 	return v.String()
 }
+
+func (*Transformer) RecordToCQTypes(table *schema.Table, record arrow.Record) (schema.CQTypes, error) {
+	data := schema.NewResourceData(table, nil, nil).ToDestinationResource().Data
+	for i, col := range record.Columns() {
+		var val any
+		if col.IsValid(0) && !col.IsNull(0) {
+			val = col.ValueStr(0)
+		}
+		// special handling for arrays
+		switch table.Columns[i].Type {
+		case schema.TypeStringArray,
+			schema.TypeIntArray,
+			schema.TypeUUIDArray,
+			schema.TypeInetArray,
+			schema.TypeCIDRArray,
+			schema.TypeMacAddrArray:
+			parts := strings.Split(strings.TrimSuffix(strings.TrimPrefix(val.(string), "["), "]"), ",")
```

Comment:
I wonder if there's a better way to read/deserialize array types from MySQL so we don't have to do this.

Reply:
me too, actually.
Still, I'd like to have @erezrokah to review this, too

Reply:
I think the issue is still caused only by tests, so the code is safe to go